.github: Add Windows GPU workflow

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58880 DO NOT MERGE, testing windows GPU workflows
* #58879 .github: Enable Windows GPU workflows on PR
* **#58782 .github: Add Windows GPU workflow**

[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

seemethere committed:
.github: Add Windows GPU workflow

[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

seemethere commented:
Tested here: https://github.com/pytorch/pytorch/pull/58880

samestep commented on .jenkins/pytorch/win-test.sh:
```diff
@@ -55,7 +55,7 @@ fi
 run_tests() {
     # Run nvidia-smi if available
-    for path in  /c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe /c/Windows/System32/nvidia-smi.exe; do
+    for path in '/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe' /c/Windows/System32/nvidia-smi.exe; do
```
What is the reason for this change?

seemethere replied:
`/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe` without the quotes represents three arguments in the for loop:

* `/c/Program`
* `Files/NVIDIA`
* `Corporation/NVSMI/nvidia-smi.exe`

Which is probably not what we're looking for since we're looking specifically for the nvidia-smi.exe binary

samestep reviewed (approved):
lgtm

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere commented on .jenkins/pytorch/win-test.sh:
```diff
@@ -55,8 +55,8 @@ fi
-    for path in  /c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe /c/Windows/System32/nvidia-smi.exe; do
-        if [ -x $path ]; then
+    for path in '/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe' /c/Windows/System32/nvidia-smi.exe; do
+        if [ -x "$path" ]; then
```
Needed for shellcheck

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

seemethere committed:
Update on ".github: Add Windows GPU workflow"


[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).
